qiskit_aer/backends/aer_compiler.py

```diff
@@ -199,12 +215,78 @@ def _inline_circuit(self, circ, continue_label, break_label, bit_map=None):
 
         return ret
 
-    def _convert_c_if_args(self, cond_tuple, bit_map):
+    class _ClbitConverter(ExprVisitor):
+        """Apply cbit index conversion in Expr tree"""
+
+        def __init__(self, bit_map):
+            self.bit_map = bit_map
+
+        def visit_value(self, node, /):
+            # pylint: disable=unused-variable
+            return node
+
+        def visit_var(self, node, /):
+            if isinstance(node.var, Clbit):
+                node.var = self.bit_map[node.var]
+            else:
+                node.var = [self.bit_map[clbit] for clbit in node.var]
+            return node
+
+        def visit_cast(self, node, /):
+            node.operand.accept(self)
+            return node
+
+        def visit_unary(self, node, /):
+            node.operand.accept(self)
+            return node
+
+        def visit_binary(self, node, /):
+            node.left.accept(self)
+            node.right.accept(self)
+            return node
+
+        def visit_generic(self, node, /):
+            raise AerError(f"unsupported expression is used: {node.__class__}")
```

Comment:
It feels like this visitor should construct the corresponding `AerExpr` tree, rather than deepcopying the Terra one and mutating it?  In particular, mutating `Var.var` is super unsafe, especially because it's being changed to a value of an incorrect type here.  (I also want to be a bit sneaky in Terra 1.0 and make `expr.Var` and `expr.Value` immutable; they logically always should have been.)

Can you re-use `_AssembleExprImpl` here?  That does the clbit remapping already, I think?

Reply:
This class is no longer used in the latest code. Now, Only `Var` instance is copied.